tests/NLog.UnitTests/Config/ConfigApiTests.cs

```diff
@@ -304,5 +313,181 @@ public void ConfigLogRuleWithName()
             Assert.Null(ruleLookup);
             Assert.False(config.RemoveRuleByName("hello"));
         }
+
+        [Fact]
+        public void LoggerNameMatcher_None()
+        {
+            var matcher = LoggerNameMatcher.Create(null);
+            Assert.Equal("logNamePattern: (:None)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_All()
+        {
+            var matcher = LoggerNameMatcher.Create("*");
+            Assert.Equal("logNamePattern: (:All)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_Empty()
+        {
+            var matcher = LoggerNameMatcher.Create("");
+            Assert.Equal("logNamePattern: (:Equals)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_Equals()
+        {
+            var matcher = LoggerNameMatcher.Create("abc");
+            Assert.Equal("logNamePattern: (abc:Equals)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_StartsWith()
+        {
+            var matcher = LoggerNameMatcher.Create("abc*");
+            Assert.Equal("logNamePattern: (abc:StartsWith)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_EndsWith()
+        {
+            var matcher = LoggerNameMatcher.Create("*abc");
+            Assert.Equal("logNamePattern: (abc:EndsWith)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_Contains()
+        {
+            var matcher = LoggerNameMatcher.Create("*abc*");
+            Assert.Equal("logNamePattern: (abc:Contains)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_MultiplePattern_StarInternal()
+        {
+            var matcher = LoggerNameMatcher.Create("a*bc");
+            Assert.Equal("logNamePattern: (^a.*bc$:MultiplePattern)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_MultiplePattern_QuestionMark()
+        {
+            var matcher = LoggerNameMatcher.Create("a?bc");
+            Assert.Equal("logNamePattern: (^a.bc$:MultiplePattern)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_MultiplePattern_EscapedChars()
+        {
+            var matcher = LoggerNameMatcher.Create("a?b.c.foo.bar");
+            Assert.Equal("logNamePattern: (^a.b\\.c\\.foo\\.bar$:MultiplePattern)", matcher.ToString());
+        }
+
+        [Fact]
+        public void LoggerNameMatcher_RegexPattern_Valid()
+        {
+            var matcher = LoggerNameMatcher.Create("{ab.c}");
```

Comment:
- this is another feature isn't? (Not described in the PR title)

- Is it's logical to wrap them in `{`? It's more common to mee to do `/ab.c/` to me (at least in Javascript ;))

Reply:
maybe in this case, a Regex property is indeed better. I was under the impression first that we didn't needed it.

Reply:
Can only repeat that I prefer a explicit regex-mode-flag instead of special pattern. Rather remove the feature than special pattern.

Reply:
I think delimiter isn't really clear, so b) disable or future or c) property 

I think c isn't difficult, but also no problems with b) as I never saw complains about the missing regex in the past years.

Reply:
> [8e417b6](https://github.com/NLog/NLog/commit/8e417b688b1e0c7d928bb7f81ae28fc509570ede) disables RegexPattern with a compiler macro `#if FEATURE_REGEXPATTERN`.
> If you prefer i can completely delete it.

I prefer deletion, in the future we can add it with another commit / branch

> Is there a place where i can see the builded documentation? I want to check how looks what i wrote....

Do you mean the sandcastle documentation? You need to run SHFB locally then (there is a config file in the repo)

Reply:
> I prefer deletion, in the future we can add it with another commit / branch

If you could create a backup in a new branch with the regex option, I could help with adding it

Reply:
Ok, i've build the doc. But there is a problem:
now `LoggerNameMatcher` is an internal class: it wraps an "internal detail" concept that isn't needed in the public scope.
But (as expected) all the documentation i wrote around `LoggerNameMatcher` class is not exposed...
Now i could:
a - publicize  `LoggerNameMatcher` with the sole purpose of publicize the doc
b - collapse all the doc in a suitable place in `LoggingRule` class (probably the `LoggerNamePattern` method with a reminder from constructors and `NameMatches` method

i prefer b. What do you think?

Reply:
Done some refinements in doc and deleted the regex option. IMHO on this PR i'm done.

Done a [branch](/beppemarazzi/NLog/tree/feature/rule-regex-pattern) with b262f87  (actually is 1 commit behind my last commit 20c4440) with regex pattern stub commented out by `#if FEATURE_REGEXPATTERN`. Or do you prefer a new PR on this?

Reply:
> Or do you prefer a new PR on this?

PR would be nice, but I think I will focus first on NLog 4.6 (rc)